lib/active_record/connection_adapters/cockroachdb/schema_statements.rb

```diff
@@ -61,15 +62,26 @@ def foreign_keys(table_name)
           SQL
 
           fk_info.map do |row|
+            to_table = PostgreSQL::Utils.unquote_identifier(row["to_table"])
+            conkey = row["conkey"].scan(/\d+/).map(&:to_i)
+            confkey = row["confkey"].scan(/\d+/).map(&:to_i)
+
+            if conkey.size > 1
+              column = column_names_from_column_numbers(row["conrelid"], conkey)
+              primary_key = column_names_from_column_numbers(row["confrelid"], confkey)
+            else
+              column = PostgreSQL::Utils.unquote_identifier(row["column"])
+              primary_key = row["primary_key"]
+            end
+
             options = {
-              column: PostgreSQL::Utils.unquote_identifier(row["column"]),
+              column: column,
               name: row["name"],
-              primary_key: row["primary_key"]
+              primary_key: primary_key
             }
-
             options[:on_delete] = extract_foreign_key_action(row["on_delete"])
             options[:on_update] = extract_foreign_key_action(row["on_update"])
-            options[:deferrable] = extract_foreign_key_deferrable(row["deferrable"], row["deferred"])
+            options[:deferrable] = extract_constraint_deferrable(row["deferrable"], row["deferred"])
```

Comment:
I'm not sure about how well `deferrable` is supported by CRDB but most tests covering it are ignored, if there is full CRDB support, we should open a related issue to track this

Reply:
CRDB doesn't support `deferrable` right now: https://github.com/cockroachdb/cockroach/issues/31632

Reply:
Perfect, that's what I understood!